### PR TITLE
fix: make sure pixi vars are available before activation.env vars are…

### PR DESCRIPTION
**Pull request by ruben-arts**
fixes: #1726 

`pixi shell-hook` before:
```bash
export ABC="$PIXI_PROJECT_ROOT/test"
...
export PIXI_PROJECT_ROOT="/home/rarts/dev/pixi"
```

`pixi shell-hook` after:
```bash
...
export PIXI_PROJECT_ROOT="/home/rarts/dev/pixi"
...
export ABC="$PIXI_PROJECT_ROOT/test"
```

### CODE REVIEW

File: src/activation.rs [hunk 1]
```diff
@@ -211,15 +214,13 @@ pub async fn run_activation(
 }
 
 /// Get the environment variables that are statically generated from the project and the environment.
+/// This returns an IndexMap to make sure they get exported in the correct order.
```
Comment (nichmor) [inline src/activation.rs:217]: wdyt about opening the parenthesis and explaining what does it mean "in the correct order"?

**Review by nichmor — COMMENTED** [review]
I think it will be good to add a small test to validate ( and show ) what correct order we expect from pixi shell-hook

File: src/activation.rs (hunk 1 above)
Comment (ruben-arts) [reply src/activation.rs:217]:
```suggestion
/// Returns IndexMap to stay sorted, as pixi should export the metadata before exporting variables that could depend on it. 
```

What about this?

Comment (nichmor) [reply src/activation.rs:217]: agree!

**Review by nichmor — APPROVED** [review]
LGTM after adding the test

**Comment by ruben-arts** [comment]
Because you @nichmor made me make a test I found another bug!

**Pull request merged**